src/main/java/newuser/NewUserInteractor.java

```diff
@@ -1,4 +1,32 @@
 package newuser;
 
-public class NewUserInteractor {
+import database.UserDbGateway;
+import entities.FacilityUser;
+import entities.User;
+import entities.FacilityType;
+
+
+public class NewUserInteractor{
+
+    UserDbGateway readWriter = new UserDbGateway();
+
+    public NewUserInteractor(){
+    }
+
+    public User addNewUser(String username, String password, int facilityID, FacilityType type){
+
+        // FacilityUser has type int facilityID, but Facility has type UUID facilityID, does this matter?
```

Comment:
FacilityUser should also be of type UUID, i think it got changed in Facility but not the FacilityUser class